test/base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/set/SetTests.kt

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.test.compiler.set
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import io.realm.kotlin.test.util.Compiler.compileFromSource
+import io.realm.kotlin.test.util.TypeDescriptor
+import io.realm.kotlin.types.RealmObject
+import org.junit.Test
+import kotlin.reflect.KClass
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+// Cannot trigger these from within the IDE due to https://youtrack.jetbrains.com/issue/KT-46195
+// Execute the tests from the CLI with `./gradlew jvmTest`
+class SetTests {
+
+    private val supportedPrimitiveTypes = TypeDescriptor.classifiers.keys.filter {
+        // Filter out RealmObject
+        it != RealmObject::class
+    }.map {
+        (it as KClass<*>).simpleName!!
+    }
+
+    private val allSupportedTypes = supportedPrimitiveTypes.plus("NonNullableSet")
+
+    // ------------------------------------------------
+    // RealmSet<E>
+    // - supported types
+    // - unsupported type fails
+    // ------------------------------------------------
+
+    @Test
+    fun `non-nullable set`() {
+        allSupportedTypes.forEach { primitiveType ->
+            val result = NON_NULLABLE_SET_CODE.format(primitiveType)
+                .let {
```

Comment:
And maybe make a wrapper method for this repeated `.let { SourceFile... }.let { compilerFromSource ... }` across the tests.